drivers/mbox/mbox_nxp_fsl_mu.c

```diff
@@ -0,0 +1,202 @@
+	}
+
+	if (enable) {
+		MU_EnableInterrupts(cfg->base,
+			(kMU_GenInt0InterruptEnable | kMU_Rx0FullInterruptEnable));
+	} else {
+		MU_DisableInterrupts(cfg->base,
+			(kMU_GenInt0InterruptEnable | kMU_Rx0FullInterruptEnable));
+	}
+
+	return 0;
+}
+
+static int nxp_fsl_mu_init(const struct device *dev)
+{
+	const struct nxp_fsl_mu_config *cfg = dev->config;
+
+	MU_Init(cfg->base);
+
+	/*
+	 * Configure and enable interrupt group, but channel's interrupt are
+	 * disabled until calling .set_enabled()
+	 */
+	cfg->config_irq();
+	return 0;
+}
+
+static const struct mbox_driver_api nxp_fsl_mu_driver_api = {
+	.send = nxp_fsl_mu_send,
+	.register_callback = nxp_fsl_mu_register_callback,
+	.mtu_get = nxp_fsl_mu_mtu_get,
+	.max_channels_get = nxp_fsl_mu_max_channels_get,
+	.set_enabled = nxp_fsl_mu_set_enabled,
+};
+
+#define MU_INIT_IRQ_FUNC(n)					\
+	void MU_##n##_IRQHandler(void);				\
+	static void nxp_fsl_mu_##n##_init_irq(void)		\
+	{							\
+		IRQ_CONNECT(DT_IRQN(MU_NODE(n)),		\
+			    DT_IRQ(MU_NODE(n), priority),	\
+			    MU_##n##_IRQHandler,		\
+			    NULL,				\
+			    0);					\
+		irq_enable(DT_IRQN(MU_NODE(n)));		\
+	}
+
+#define MU_INSTANCE_DEFINE(n)							\
+	static struct nxp_fsl_mu_data nxp_fsl_mu_##n##_data;			\
+	static struct nxp_fsl_mu_config nxp_fsl_mu_##n##_config = {		\
+		.base = MU_BASE(n),						\
+		.config_irq = nxp_fsl_mu_##n##_init_irq,			\
+	};									\
+										\
+	DEVICE_DT_DEFINE(MU_NODE(n), nxp_fsl_mu_init, NULL,			\
+			&nxp_fsl_mu_##n##_data, &nxp_fsl_mu_##n##_config,	\
+			POST_KERNEL, CONFIG_MBOX_INIT_PRIORITY,			\
+			&nxp_fsl_mu_driver_api)
+
+#define MU_IRQ_HANDLER(n)							\
+	static uint32_t mu_##n##_received_data;					\
+	void MU_##n##_IRQHandler(void)						\
+	{									\
+		const struct device *dev = DEVICE_DT_GET(MU_NODE(n));		\
+		const struct nxp_fsl_mu_data *data = dev->data;			\
+		const struct nxp_fsl_mu_config *config = dev->config;		\
+		int channel = 0;						\
+		struct mbox_msg msg;						\
+		struct mbox_msg *callback_msg_ptr = NULL;			\
+		uint32_t flag = MU_GetStatusFlags(config->base);		\
+										\
+		if ((flag & kMU_Rx0FullFlag) == kMU_Rx0FullFlag) {		\
+			mu_##n##_received_data =				\
+				MU_ReceiveMsgNonBlocking(config->base, 0);	\
+			msg.data = (const void *)&mu_##n##_received_data;	\
+			msg.size = MU_MBOX_SIZE;				\
+			callback_msg_ptr = &msg;				\
+		} else if ((flag & kMU_GenInt0Flag) == kMU_GenInt0Flag) {	\
+			MU_ClearStatusFlags(config->base, kMU_GenInt0Flag);	\
+			callback_msg_ptr = NULL;				\
+		}								\
+										\
+		if (data->cb[channel]) {					\
+			data->cb[channel](dev, channel,				\
+					data->user_data[channel],		\
+					callback_msg_ptr);			\
+		}								\
+	}
+
+#if DT_NODE_HAS_STATUS(MU_NODE(0), okay)
+MU_INIT_IRQ_FUNC(0);
+MU_INSTANCE_DEFINE(0);
+MU_IRQ_HANDLER(0);
+#endif
+
+#if DT_NODE_HAS_STATUS(MU_NODE(1), okay)
+MU_INIT_IRQ_FUNC(1);
+MU_INSTANCE_DEFINE(1);
+MU_IRQ_HANDLER(1);
+#endif
```

Comment:
Why not using something more generic like `DT_INST_FOREACH_STATUS_OKAY`?

Reply:
Done